projects/self-creation-portal/src/app/services/common-service/common.service.ts

```diff
@@ -53,7 +53,17 @@ export class CommonService {
 
   clearQueryParams() {
     const queryParams = this.route.snapshot.queryParams;
-    if (Object.keys(queryParams).length > 0) {
+    const currentPath = this.router.url.split('?')[0];
+    const sidenavPaths = [
+      '/home/drafts',
+      '/home/up-for-review',
+      '/home/submitted-for-review'
+    ];
+    // Check if the current path matches any of the tab paths, if matches then clear params and redirect to route
+    const issidenavPaths = sidenavPaths.some(tabPath => currentPath.startsWith(tabPath));
+  
+    if ((currentPath.includes('solution') && issidenavPaths && Object.keys(queryParams).length > 0) || 
```

Comment:
Can you explain "solution" string comparision

Reply:
as it is related to this url "/solution/project/project-details" then it should not clear the params and getback to original route.
when searching after some searchtext and clicking on the button then ngDestroy detecting it coming from different component so to handle that case I have tried here only when this particular sidenavpaths are there then only clearparams else don't clear, so to fixed this issue I am checking currentpath includes solutions url.

Reply:
The please define as parent route and compare here, don't hard code